### PR TITLE
Azure reference architecture

### PR DESCRIPTION
**Pull request by ned1313**
This adds a reference architecture in Azure for Boundary. Confirmed to work with Boundary 0.1.7.

### CODE REVIEW

File: deployment/azure/azure/boundary.tmpl [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+# Create controller config if type is "controller"
+# This config assumes you are using postgresql as the database and 
+# Azure Key Vault for the KMS
+# This config also assumes that you have are using a certificate created by Azure Key Vault
+# or you could source it another way
+#
+# TODO: should mlock be disabled?
+%{ if type == "controller"}
+cat <<EOF > ~/boundary-controller.hcl
+disable_mlock = true
```
Comment (malnick) [inline deployment/azure/azure/boundary.tmpl:43]: https://www.boundaryproject.io/docs/configuration#disable_mlock

I think for this example it's fine to leave it disabled, you'll need extra systemd directives if you enable it.

**Review by malnick — COMMENTED** [review]
This looks really good! Thank you so much for contributing. I had some comments, but after those are responded to, happy to move this to approved.

Thanks again!

File: deployment/azure/README.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+## Deploy
+To deploy this example:
+- Go into the `/azure` subdirectory
+- Ensure you are logged into Azure (`az login`)
+- Run the following:
+
+```
+terraform init
+terraform apply
+```
```
[the remark this answers is not in the document]
Comment (ned1313) [reply deployment/azure/README.md:17]: Changed it to be consistent with the AWS example.

File: deployment/azure/azure/boundary.tmpl (hunk 1 above)
Comment (ned1313) [reply deployment/azure/azure/boundary.tmpl:42]: Removed the comment.

**Review by malnick — APPROVED** [review]

**Pull request merged**